### PR TITLE
Add "incognito" fava-option to hide digits

### PR DESCRIPTION
**Pull request by aumayr**
Follow-up to PR #457, see discussion there.

### CODE REVIEW

**Comment by aumayr** [comment]
This does not implement a command line option when starting Fava, as I do not know how to elegantly pass the option from `cli.py` to the app object.

File: fava/application.py [hunk 1]
```diff
@@ -159,6 +160,19 @@ def _perform_global_filters():
         flash(str(exception))
 
 
+@app.after_request
+def _incognito(response):
+    """Replace all numbers with 'X'."""
+    if (g.ledger.fava_options['incognito'] and
```
Comment (yagebu) [inline fava/application.py]: this shouldn't fail if `g.ledger` doesn't exist

**Comment by yagebu** [comment]
A test would be nice

File: tests/test_application.py [hunk 2]
```diff
@@ -76,3 +76,14 @@ def test_jump_handler(app, test_client, referer, jump_link, expect):
             expect)
         assert result.status_code == 302
         assert get_url == expect_url
+
+
+def test_incognito(app, test_client):
+    with app.test_request_context():
+        app.preprocess_request()
+        flask.g.ledger.fava_options['incognito'] = True
```
Comment (yagebu) [inline tests/test_application.py]: This should probably be reset after this test as the ledger object will be shared between tests

**Pull request merged**